src/Smalot/PdfParser/Page.php

```diff
@@ -240,6 +240,12 @@ public function getTextArray(self $page = null): array
 
                     $header = new Header([], $this->document);
                     $contents = new PDFObject($this->document, $header, $new_content, $this->config);
+                } else {
+                    try {
+                        $contents->getTextArray($this);
+                    } catch (\Error $e) {
```

Comment:
Why not `Exception` here?

Reply:
Because it raises (throw) and Error and not an Exception. (I don't know why it throw Error instead of an Exception). So catching an Exception will not help because Error class does not inherit from Exception:
`class Error implements Throwable` (This is from the php online docs in https://www.php.net/manual/en/class.error.php)

Actually in the php online documentations: https://www.php.net/manual/en/language.errors.php7.php
It says: 
_**As the Error hierarchy does not inherit from Exception, code that uses catch (Exception $e) { ... } blocks to handle uncaught exceptions in PHP 5 will find that these Errors are not caught by these blocks. Either a catch (Error $e) { ... } block or a set_exception_handler() handler is required.**_
And even though it says that the problem is in PHP 5, I tested it... I am using PHP 8 and it does not catch it, unless you catch Error class.

Reply:
Let me know if my answer is OK.

Reply:
I would catch `Throwable` then. In both cases the semantics behind is `Something went wrong`, isn't it?

Reply:
Good! Let me do that... and test!!!

Reply:
Good advise @k00ni! Done!